Code/Framework/AzFramework/Platform/Android/AzFramework/IO/LocalFileIO_Android.cpp

```diff
@@ -155,43 +144,35 @@ namespace AZ
 
         Result LocalFileIO::CreatePath(const char* filePath)
         {
-            char resolvedPath[AZ::IO::MaxPathLength];
-            ResolvePath(filePath, resolvedPath, AZ::IO::MaxPathLength);
+            FixedMaxPath resolvedPath;
+            ResolvePath(resolvedPath, filePath);
 
-            if (AZ::Android::Utils::IsApkPath(resolvedPath))
+            if (AZ::Android::Utils::IsApkPath(resolvedPath.c_str()))
             {
                 return ResultCode::Error; //you can't write to the APK
             }
 
             // create all paths up to that directory.
             // its not an error if the path exists.
-            if ((Exists(resolvedPath)) && (!IsDirectory(resolvedPath)))
+            if (Exists(resolvedPath.c_str()) && !IsDirectory(resolvedPath.c_str()))
             {
                 return ResultCode::Error; // that path exists, but is not a directory.
             }
 
             // make directories from bottom to top.
-            AZStd::string pathBuffer;
-            size_t pathLength = strlen(resolvedPath);
-            pathBuffer.reserve(pathLength);
-            for (size_t pathPos = 0; pathPos < pathLength; ++pathPos)
+            FixedMaxPath directoryPath = resolvedPath.RootPath();
+            for (AZ::IO::PathView pathSegment : resolvedPath.RelativePath())
             {
-                if ((resolvedPath[pathPos] == '\\') || (resolvedPath[pathPos] == '/'))
+                directoryPath /= pathSegment;
+                mkdir(pathBuffer.c_str(), S_IRWXU | S_IRWXG | S_IROTH | S_IXOTH);
+                if (!IsDirectory(pathBuffer.c_str()))
                 {
-                    if (pathPos > 0)
-                    {
-                        mkdir(pathBuffer.c_str(), S_IRWXU | S_IRWXG | S_IROTH | S_IXOTH);
-                        if (!IsDirectory(pathBuffer.c_str()))
-                        {
-                            return ResultCode::Error;
-                        }
-                    }
+                    return ResultCode::Error;
                 }
-                pathBuffer.push_back(resolvedPath[pathPos]);
             }
 
-            mkdir(pathBuffer.c_str(), S_IRWXU | S_IRWXG | S_IROTH | S_IXOTH);
-            return IsDirectory(resolvedPath) ? ResultCode::Success : ResultCode::Error;
+            mkdir(directoryPath.c_str(), S_IRWXU | S_IRWXG | S_IROTH | S_IXOTH);
```

Comment:
should we mkdir before loop at first? as original implementation mkdir before appending path in the loop